wagtail_wordpress_import/test/tests/test_shortcode_handlers.py

```diff
@@ -236,62 +248,59 @@ def test_included_shortcodes(self):
         # note this class has not been registered
         class FooHandler(BlockShortcodeHandler):
             shortcode_name = "foo"
+            custom_html_tag_prefix = "wagtail_block_"
 
-        registered_handlers = SHORTCODE_HANDLERS.keys()
-        self.assertIn("caption", registered_handlers)
-        self.assertNotIn("foo", registered_handlers)
+        caption_handler_instance = SHORTCODE_HANDLERS[0]()
+
+        self.assertIsInstance(caption_handler_instance, CaptionHandler)
+        self.assertNotIsInstance(caption_handler_instance, FooHandler)
+        self.assertEqual(len(SHORTCODE_HANDLERS), 1)
 
 
 class TestIncludedShortcodeHandlers(TestCase):
     def test_included_shortcodes(self):
         # prime the SHORTCODE_HANDLERS
         # note this class has been registered
-        @register("foo")
+        @register()
         class FooHandler(BlockShortcodeHandler):
             shortcode_name = "foo"
+            custom_html_tag_prefix = "wagtail_block_"
+
+        caption_handler_instance = SHORTCODE_HANDLERS[0]()
+        foo_handler_instance = SHORTCODE_HANDLERS[1]()
 
-        registered_handlers = SHORTCODE_HANDLERS.keys()
-        self.assertIn("foo", registered_handlers)
-        self.assertIn("caption", registered_handlers)
+        self.assertIsInstance(caption_handler_instance, CaptionHandler)
+        self.assertIsInstance(foo_handler_instance, FooHandler)
+        self.assertEqual(len(SHORTCODE_HANDLERS), 2)
 
 
 class TestShortcodeHandlerStreamfieldBlockCreation(TestCase):
     def test_construct_block_method_output(self):
         handler = CaptionHandler()
-        wagtail_custom_html = (
-            '<wagtail_block_caption id="attachment_46162" align="aligncenter" width="600">'
-            '<img class="wp-image-46162 size-full" '
-            'src="https://www.example.com/images/foo.jpg" '
+        wagtail_custom_html = BeautifulSoup(
+            '<wagtail_block_caption id="46162" align="aligncenter" width="600">'
+            '<img class="size-full" '
+            'src="https://www.budgetsaresexy.com/images/j-money-family-portrait.jpg" '
```

Comment:
**Issue**: We have a remote URL here. Are we querying this during the tests?

Reply:
Yes that gets queried here: https://github.com/torchbox/wagtail-wordpress-import/blob/77ad1c0aa3a1ec41c829bb2d5b89631f97727391/wagtail_wordpress_import/block_builder_defaults.py#L222 

It's function that's also used when linking rich_text images.

We have a ticket to make this better in testing: https://projects.torchbox.com/projects/wordpress-to-wagtail-importer-package/tickets/76